### PR TITLE
Validate the param of rand function in compile step

### PR DESCRIPTION
**Pull request by xinghuayu007**
## Proposed changes

The param of **rand()** function should be literal, but current compiler ingore to validate the literal param of rand function, it is validated in execution step. This PR make it validated in compile step, anbd make it more earlier to find the usage error of **rand()** function.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [] I have create an issue on (Fix #ISSUE), and have described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If this change need a document change, I have updated the document
- [x] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/analysis/Expr.java [hunk 1]
```diff
@@ -1405,7 +1405,15 @@ protected Function getBuiltinFunction(
             throws AnalysisException {
         FunctionName fnName = new FunctionName(name);
         Function searchDesc = new Function(fnName, argTypes, Type.INVALID, false);
-        return Catalog.getCurrentCatalog().getFunction(searchDesc, mode);
+        Function f = Catalog.getCurrentCatalog().getFunction(searchDesc, mode);
+        if (f != null && fnName.getFunction().equalsIgnoreCase("rand")) {
+            if (this.children != null
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/Expr.java:1410]: `this.children != null` is unnecessary

File: fe/fe-core/src/test/java/org/apache/doris/analysis/ExprTest.java [hunk 2]
```diff
@@ -24,18 +24,43 @@
...
+import java.io.IOException;
+import java.util.*;
 
 import mockit.Expectations;
 import mockit.Injectable;
 import mockit.Mocked;
 
 public class ExprTest {
 
+    private static String runningDir = "fe/mocked/DemoTest/" + UUID.randomUUID().toString() + "/";
```
Comment (kangkaisen) [inline fe/fe-core/src/test/java/org/apache/doris/analysis/ExprTest.java:46]: You could add the test in `SelectStmtTest`, and then you could directly write `testRandFunction `

File: fe/fe-core/src/test/java/org/apache/doris/analysis/ExprTest.java [hunk 3]
```diff
@@ -159,4 +184,42 @@ public void testUncheckedCastTo() throws AnalysisException {
...
+        }
+
+        try {
+            ConnectContext ctx = UtFrameUtils.createDefaultCtx();
+            String selectStmtStr = "select rand(1234) from db1.tbl1;";
+            UtFrameUtils.parseAndAnalyzeStmt(selectStmtStr, ctx);
+        } catch (Exception e) {
+            Assert.assertFalse(e.getMessage().contains("The param of rand function must be literal"));
+        }
+        Assert.assertTrue(1 == 1);
```
Comment (kangkaisen) [inline fe/fe-core/src/test/java/org/apache/doris/analysis/ExprTest.java:205]: What's the meaning of `  Assert.assertTrue(1 == 1);` ?

File: fe/fe-core/src/main/java/org/apache/doris/analysis/Expr.java (hunk 1 above)
Comment (xinghuayu007) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/Expr.java:1410]: Rand function has two implements. One is Rand(const int), the other is Rand() without parm**.  this.children != null** filters Rand() without param.

File: fe/fe-core/src/test/java/org/apache/doris/analysis/SelectStmtTest.java [hunk 4]
```diff
@@ -362,4 +362,39 @@ public void testDataGripSupport() throws Exception {
                 "from information_schema.collations";
         dorisAssert.query(sql).explainQuery();
     }
+
+    @Test
+    public void testRandFunction() {
+        try {
+            ConnectContext ctx = UtFrameUtils.createDefaultCtx();
```
Comment (kangkaisen) [inline fe/fe-core/src/test/java/org/apache/doris/analysis/SelectStmtTest.java:369]: You could use `dorisAssert.query(sql).explainQuery();`, which is more simpler.

File: fe/fe-core/src/main/java/org/apache/doris/analysis/Expr.java (hunk 1 above)
Comment (kangkaisen) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/Expr.java:1410]: children is init with `Lists.newArrayList();` in `TreeNode`, so it won't be null.

Comment (xinghuayu007) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/Expr.java]: You are right.

**Review by kangkaisen — DISMISSED** [review]
+1

**Comment by morningman** [comment]
Hi @xinghuayu007 , you have to rebase to master to solve the conflict.

**Comment by xinghuayu007** [comment]
> Hi @xinghuayu007 , you have to rebase to master to solve the conflict.

done

**Review by kangkaisen — APPROVED** [review]
+1

**Pull request merged**